hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/utils/db/RDBTable.java

```diff
@@ -125,8 +127,18 @@ public boolean isExist(byte[] key) throws IOException {
       // RocksDB#keyMayExist
       // If the key definitely does not exist in the database, then this
       // method returns false, else true.
-      return db.keyMayExist(handle, key, new StringBuilder())
-          && db.get(handle, key) != null;
+      rdbMetrics.incNumDBKeyMayExistChecks();
+      StringBuilder outValue = new StringBuilder();
+      boolean keyMayExist = db.keyMayExist(handle, key, outValue);
+      if (keyMayExist) {
+        boolean keyExists = (outValue.length() > 0) ||
```

Comment:
Could you explain that why you need the addition condition to judge the value of keyMayExist?